File: Arrays/3430-count-days-without-meetings/count-days-without-meetings.java
class Solution {
    public int countDays(int days, int[][] meetings) {
        Arrays.sort(meetings, (a, b) -> Integer.compare(a[0], b[0]));

        int totalFreeDays = 0;
        int lastEnd = 0;

        for (int[] meeting : meetings) {
            if (meeting[0] > lastEnd) {
                totalFreeDays += meeting[0] - lastEnd - 1;
            }
            lastEnd = Math.max(lastEnd, meeting[1]);
        }

        totalFreeDays += days - lastEnd;
        return totalFreeDays;
    }
}

File: Arrays/3430-count-days-without-meetings/count-days-without-meetings.cpp
class Solution {
public:
    int countDays(int days, vector<vector<int>>& meetings) {
        sort(meetings.begin(), meetings.end());

        int sum = 0;
        int last_end = 0;

        for(auto meeting:meetings){
            if(meeting[0] > last_end) sum += meeting[0] - last_end - 1;

            last_end = max(last_end, meeting[1]);
        }

        sum += days - last_end;

        return sum;
    }
};